### PR TITLE
Add pages title

### PR DESCRIPTION
**Pull request by montogeek**
Implements #97, this PR represents an original approach.

About the title format, I propose:
Homepage: ReasonML
Top level pages: {pageName} - ReasonML. Example: Playground - ReasonML
Docs pages: {method} - {nameSpace} - ReasonML. Example: Global - JS - ReasonML.

Please let me know your thoughts.

### CODE REVIEW

File: components/Meta.re [hunk 1]
```diff
@@ -1,12 +1,14 @@
 module Head = Next.Head;
+open Util.ReactStuff;
 
 [@react.component]
-let make = () => {
+let make = (~title) => {
```
Comment (montogeek) [inline components/Meta.re]: Maybe we can make it optional.

File: layouts/JsDocsLayout.re [hunk 2]
```diff
@@ -236,10 +236,11 @@ module Docs = {
       theme=`Js
       components
       sidebar
-      breadcrumbs=?breadcrumbs
+      ?breadcrumbs
       route={
         router##route;
       }>
+      <Meta title={moduleName ++ " - ReasonML"} />
```
Comment (montogeek) [inline layouts/JsDocsLayout.re]: No 100% sure about if this is the correct place to call this component, I would prefer to do:
```js
<>
 <Meta />
 ...rest of components
<>
```

File: scripts/extract-indices.js [hunk 3]
```diff
@@ -97,19 +97,19 @@ const createIndex = result => {
...
 };
 
-const BELT_MD_DIR = path.join(__dirname, "../pages/apis/javascript/latest/belt");
+const BELT_MD_DIR = path.join(__dirname, "../pages/apis/javascript/latest/");
 const BELT_INDEX_FILE = path.join(
   __dirname,
   "../index_data/belt_api_index.json"
 );
-const beltFiles = glob.sync(`${BELT_MD_DIR}/*.md?(x)`);
+const beltFiles = glob.sync(`${BELT_MD_DIR}{belt/*.md?(x),belt.mdx}`);
```
Comment (montogeek) [inline scripts/extract-indices.js]: This change is to put the index files of `belt` and `js` in the index, so we have a title for the page title.

**Comment by ryyppy** [comment]
Since we introduced frontmatter support with #135, i guess this PR will be stale... I will close this for now, feel free to reopen another one if it's still relevant!

**Pull request closed without merging**